### PR TITLE
[Merged by Bors] - chore: deprecate `Ordering.orElse` for `.then`

### PR DESCRIPTION
**Pull request by digama0**
`Ordering.orElse` is a duplicate of `Ordering.then` from Init. Similarly, `toRel` is a duplicate of `Compares`. (`cmp` is also a duplicate of `compareOfLessAndEq`, but it's used quite heavily and the new name is a bit...)

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Init/Data/Ordering/Basic.lean [hunk 1]
```diff
@@ -14,18 +16,28 @@ deriving instance Repr for Ordering
...
+  | gt, a, b => a > b
+#align ordering.compares Ordering.Compares
+
+@[deprecated] alias toRel := Compares
+
+@[simp] lemma compares_lt [LT α] (a b : α) : Compares lt a b = (a < b) := rfl
+
+@[simp] lemma compares_eq [LT α] (a b : α) : Compares eq a b = (a = b) := rfl
+
+@[simp] lemma compares_gt [LT α] (a b : α) : Compares gt a b = (a > b) := rfl
```
Comment (eric-wieser) [inline Mathlib/Init/Data/Ordering/Basic.lean:40]: Do we actually want to be moving things *into* `Init`? My impression was that `Init` was basically just "cruft from core that we haven't decided what to do with yet", for which the natural progression is to gradually either promote or delete declarations.

Comment (digama0) [reply Mathlib/Init/Data/Ordering/Basic.lean:40]: I did this only from a avoiding-unnecessary-imports perspective. `Compares` was in a file with lots of imports and there are other `Init` files that import this and use `toRel`, now `Compares`.

**Comment by grunweg** [comment]
The file in this PR was moved out of Init - so it seems this PR is unblocked.
@digama0 Can you merge master, please - so this PR can get reviewed? Thanks!

**Review by eric-wieser — APPROVED** [review]
bors d+

Thanks!

**Pull request closed without merging**